controller_manager:
  ros__parameters:
    update_rate: 100  # Hz

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    husky_velocity_controller:
      type: diff_drive_controller/DiffDriveController

    panda_arm_controller:
      type: joint_trajectory_controller/JointTrajectoryController

    panda_hand_controller:
      type: position_controllers/GripperActionController

husky_velocity_controller:
  ros__parameters:
    left_wheel_names: ["front_left_joint", "back_left_joint"]
    right_wheel_names: ["front_right_joint", "back_right_joint"]
    wheel_separation: 0.57
    wheel_radius: 0.177
    use_stamped_vel: false

panda_arm_controller:
  ros__parameters:
    joints:
      - panda_joint1
      - panda_joint2
      - panda_joint3
      - panda_joint4
      - panda_joint5
      - panda_joint6
      - panda_joint7
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity

panda_hand_controller:
  ros__parameters:
    joint: panda_finger_joint1
    allow_stalled: true